### ISSUE TITLE
Set-AzureVMChefExtension fails for azure-powershell v3.1.0

### ISSUE THREAD

**Issue by NimishaS** [Azure/azure-powershell#3267]
### Cmdlet(s)
```
Set-AzureVMChefExtension -ValidationPem D:\chef-repo\.chef\nimisha-validator.pem -ClientRb C:\Users\Administrator\Desktop\client.rb -Linux -VM $vmm
```

### PowerShell Version
4.0

### OS Version
6.3.9600.16406

### Description
https://github.com/chef-partners/azure-chef-extension/issues/179

### Script/Steps for Reproduction
```
$vm1 = "vm-name"
$svc = "vm-name"
$vmm = Get-AzureVM -Name $vm1 -ServiceName $svc
$vmOb = Set-AzureVMChefExtension -ValidationPem D:\chef-repo\.chef\nimisha-validator.pem -ClientRb C:\Users\Administrator\Desktop\client.rb -Linux -VM $vmm
Update-AzureVM -VM $vmOb.VM -Name $vm1 -ServiceName $svc
```

**Comment by NimishaS** [comment]
@markcowl , @shahabhijeet, I tested with `azure-powershell` v1.x and 3.1.0 and I could reproduce this issue. But the command is working successfully with `azure-powershell` master branch.
I think the changes of this PR: https://github.com/Azure/azure-powershell/pull/3083 need to be released.
@Vasu1105 had made changes to pass `PublicConfiguration` as HashTable. Previously it was being passed as String.

**Comment by NimishaS** [comment]
@markcowl , @shahabhijeet, any updates on this?

**Comment by cormacpayne** [comment]
@NimishaS it looks like PR https://github.com/Azure/azure-powershell/pull/3083 was merged earlier this month, so it should be available in PowerShell 3.3.0

**Comment by NimishaS** [comment]
Thanks @cormacpayne. The command is working fine now.
